dojo/unittests/tools/test_nuclei_parser.py

```diff
@@ -0,0 +1,112 @@
+            self.assertEqual("nuclei-example.com", finding.unsaved_endpoints[0].host)
+            self.assertEqual(443, finding.unsaved_endpoints[0].port)
+
+        with self.subTest(i=3):
+            finding = findings[3]
+            self.assertEqual("Wappalyzer Technology Detection", finding.title)
+            self.assertEqual("Info", finding.severity)
+            self.assertEqual(11, finding.nb_occurences)
+            self.assertIsNotNone(finding.description)
+            self.assertIsNone(finding.references)
+            self.assertIsNone(finding.unsaved_tags)
+            self.assertEqual("WebGoat", finding.unsaved_endpoints[0].path)
+            self.assertEqual("127.0.0.1", finding.unsaved_endpoints[0].host)
+            self.assertEqual(8080, finding.unsaved_endpoints[0].port)
+            self.assertEqual("WebWolf", finding.unsaved_endpoints[1].path)
+            self.assertEqual("127.0.0.1", finding.unsaved_endpoints[1].host)
+            self.assertEqual(9090, finding.unsaved_endpoints[1].port)
+            self.assertEqual(None, finding.unsaved_endpoints[2].path)
+            self.assertEqual("nuclei-example.com", finding.unsaved_endpoints[2].host)
+            self.assertEqual(443, finding.unsaved_endpoints[2].port)
+
+        with self.subTest(i=4):
+            finding = findings[4]
+            self.assertEqual("WAF Detection", finding.title)
+            self.assertEqual("Info", finding.severity)
+            self.assertEqual(2, finding.nb_occurences)
+            self.assertIsNotNone(finding.description)
+            self.assertIsNone(finding.references)
+            self.assertIsNone(finding.unsaved_tags)
+            self.assertEqual(None, finding.unsaved_endpoints[0].path)
+            self.assertEqual("nuclei-example.com", finding.unsaved_endpoints[0].host)
+            self.assertEqual(443, finding.unsaved_endpoints[0].port)
+
+        with self.subTest(i=5):
+            finding = findings[5]
+            self.assertEqual("phpMyAdmin Panel", finding.title)
+            self.assertEqual("Info", finding.severity)
+            self.assertEqual(1, finding.nb_occurences)
+            self.assertIsNotNone(finding.description)
+            self.assertIsNone(finding.references)
+            self.assertEqual("panel", finding.unsaved_tags)
+            self.assertEqual("phpmyadmin/", finding.unsaved_endpoints[0].path)
+            self.assertEqual("nuclei-example.com", finding.unsaved_endpoints[0].host)
+            self.assertEqual(443, finding.unsaved_endpoints[0].port)
+
+        with self.subTest(i=6):
+            finding = findings[6]
+            self.assertEqual("MySQL DB with enabled native password", finding.title)
+            self.assertEqual("Info", finding.severity)
+            self.assertEqual(1, finding.nb_occurences)
+            self.assertIsNotNone(finding.description)
+            self.assertIsNone(finding.references)
+            self.assertEqual("network,mysql,bruteforce,db", finding.unsaved_tags)
```

Comment:
unsaved tags should be an array.
Also better to have checks like that:
```python
    self.assertIn("network", finding.unsaved_tags)
    self.assertIn("mysql", finding.unsaved_tags)
```

Reply:
Done